src/lib.rs

```diff
@@ -1,40 +1,97 @@
-extern crate byteorder;
-#[macro_use]
-extern crate enum_primitive;
-extern crate num;
+use std::any::type_name;
+use std::io::{Read, Seek};
 
-use std::fmt;
-use std::io::Result;
+use num_traits::FromPrimitive;
+use tiff::decoder::{Decoder, DecodingResult};
+use tiff::tags::Tag;
+use tiff::TiffResult;
 
-pub mod geotiff;
-mod lowlevel;
-mod reader;
+use crate::image_data::*;
 
-pub use geotiff::TIFF;
-use reader::*;
+mod image_data;
 
-/// The GeoTIFF library reads `.tiff` files.
+macro_rules! assert_type {
+    ($result: expr, $actual: ty, $expected: ty) => {
```

Comment:
This macro is :fire:! Could we rename it from `assert_type` to something else though? I realize it does do a check/assertion, but it's purpose is more about converting `ImageData` to a primitive type, so maybe `to_primitive_type` or something like that?

Reply:
I agree; I was a bit lazy on this one. Since the macro is not doing the conversion but just unpacking the result if the type is correct, would maybe something like `unwrap_primitive_type` be more appropriate?

Reply:
Yep, we can go with `unwrap_primitive_type`, though we might remove the `unwrap` call at some point :wink: